docs/services/sdk/error-handling.mdx

```diff
@@ -0,0 +1,68 @@
+---
+sidebar_position: 8
+description: "Restate's retry strategy and how to manage it from the SDK."
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Error handling
+
+Restate takes care of retries of failed invocations. 
+It does this for all kinds of errors, such as transient network errors and errors thrown inside handlers. 
+By default, Restate does infinite retries with an exponential backoff strategy.
+
+For situations where you do not want the invocation to be retried, but instead you want the error message
+to be propagated back to the caller, you can use a terminal error.
+
+<Tabs groupId="sdk">
+<TabItem value="ts" label="Typescript">
+
+```typescript
+import { TerminalError } from "@restatedev/restate-sdk";
+
+// Inside your handler throw terminal errors via:
+throw new TerminalError("Something went wrong.", { errorCode: 13, cause: "Something caused this." })
+```
+
+The `errorCode` and `cause` are optional.
+
+:::note Errors during side effect execution
+
+Throwing errors inside a side effect has the same semantics as anywhere else in your handler code.
+By default, Restate infinitely retries the invocation, and therefore the side effect.
+Errors that should not be retried, should be thrown as terminal errors, as described earlier.
+
+Invocation retries are triggered by Restate, not by the SDKs.
+For some use cases, you may want to do the side effect retries from within the invocation,
+because this is faster and more efficient in certain scenarios (e.g. you have a connection set up to an external system or database).
+Have a look at the [side effect documentation](https://docs.restate.dev/services/sdk/side-effects#retrying-on-failure)
+on how to set this up.
+:::
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+
+You can throw a terminal exception via `dev.restate.sdk.core.TerminalException` and supply it with a `dev.restate.sdk.core.TerminalException.Code` and a message:
+
+```java
+import dev.restate.sdk.core.TerminalException;
+
+// Inside your handler throw terminal errors with a code and message as follows:
+throw new TerminalException(Code.UNKNOWN, "Something went wrong");
+```
+
+:::note Errors during side effect execution
+Throwing errors inside a side effect has the same semantics as anywhere else in your handler code.
+By default, Restate infinitely retries the invocation, and therefore the side effect.
+Errors that should not be retried, should be thrown as terminal errors, as described earlier.
+Invocation retries are triggered by Restate, not by the SDKs.
+:::
+
```

Comment:
This text is missing to describe that you can catch the `TerminalException`s, for example the one coming out of `call().await()`, to build your control flow.